uis/src/com/biglybt/ui/swt/columns/torrent/ColumnProgressETA.java

```diff
@@ -493,6 +492,14 @@ public void cellPaint(GC gc, TableCellSWT cell) {
 
 		boundsProgressBar.width--;	// adjust as includes pixel for column line :(
 		
+		final int lineCount = cell.getMaxLines();
+		int lineHeight;
+		if ( lineCount > 1 ) {
+			lineHeight = cell.getTableRow().getView().getLineHeight();
+		} else {
+			lineHeight = cellBounds.height;
```

Comment:
This won't interfere with ProgressETA activating two-line mode when it can, even in single-text-line height rows. It initially seems like it would, and it took me a bit to convince myself it wouldn't.

It **seems** like a mistake that the code won't do the `lineHeight` lookup from the view unless `.getMaxLines()` is at least 2, but that's actually correct. It doesn't need the real line height in those cases. In fact, that's the whole point of this `Math.Min()` call: ` Math.min( minTwoLineHeight, lineHeight*2 )` .

Unless we can fit 2+ lines of text in the row, we want the `showSecondLine` decision to be based on `minTwoLineHeight`. Using a `lineHeight` equal to the cell height ensures that it will be. Whether that's because the row only fits one actually-rendered line of text at the given font size, or because it's a FakeTableCell with no real height at all, doesn't matter to the code here.